test/e2e_new/features/kafka_source.go

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2023 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package features
+
+import (
+	"fmt"
+
+	"knative.dev/eventing-kafka/test/rekt/resources/kafkasource"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/reconciler-test/pkg/feature"
+	"knative.dev/reconciler-test/resources/svc"
+
+	testingpkg "knative.dev/eventing-kafka-broker/test/pkg"
+)
+
+func SetupAndCleanupKafkaSources(n int) *feature.Feature {
+	f := feature.NewFeatureNamed("setup and cleanup KafkaSources")
+
+	prefix := "kafka-source-"
+	sink := "sink"
+
+	f.Setup("install a sink", svc.Install(sink, "app", "rekt"))
+
+	for i := 0; i < n; i++ {
+		topicName := feature.MakeRandomK8sName("topic") // A k8s name is also a valid topic name.
```

Comment:
is the kafkatopic being created anywhere?

Reply:
No, good catch!